internal/serve/httphandler/disbursement_handler.go

```diff
@@ -283,6 +278,32 @@ func (d DisbursementHandler) PostDisbursementInstructions(w http.ResponseWriter,
 	httpjson.Render(w, response, httpjson.JSON)
 }
 
+// parseCsvFromMultipartRequest parses the CSV file from a multipart request and returns the file content and header,
+// or an error if the file is not a valid CSV or the MIME type is not text/csv.
+func parseCsvFromMultipartRequest(r *http.Request) (*bytes.Buffer, *multipart.FileHeader, *httperror.HTTPError) {
+	// Parse uploaded CSV file
+	file, header, err := r.FormFile("file")
+	if err != nil {
+		return nil, nil, httperror.BadRequest("could not parse file", err, nil)
+	}
+	defer file.Close()
```

Comment:
Is it worth to add a check for MIME type ? 

```golang
if header.Header.Get("Content-Type") != "text/csv" {
    return nil, nil, httperror.BadRequest("the MIME type should be text/csv", nil, nil)
}
```

Reply:
The content type will be `multipart/form-data` here. Also, the mime type is inferred by the gocsv parser that validates if the csv is in a valid format.